### PR TITLE
Add daemon mode continue (WIP)

### PR DESCRIPTION
**Pull request by breml**
@magnusbaeck I continued to work on the daemon mode. There are still quite some issues to be resolved and features to be added. Feel free to have a look. If you do review this, give me a shout, such that I can continue on a different branch with the further development.

The most interesting features, that have been added are:

* Support for codecs in input plugin
* Support multiple inputs
* Make the @metadata key configurable

Additionally some bug fixing and small improvements here and there.

I also updated the checklist in https://github.com/magnusbaeck/logstash-filter-verifier/issues/96#issuecomment-797559730

### CODE REVIEW

**Comment by magnusbaeck** [comment]
Starting to look at this now. Should I go ahead and push commits that look fine into master as I go along? Will obviously only work until I have comments for a commit, but still.

**Comment by breml** [comment]
As you like. I forked a new branch for my current work. If you have comments, I will decide if I fix them on the PR or on my new branch depending on complexity and effort. When ever this PR is merged, I will rebase my new branch on master before creating a new PR.

**Review by magnusbaeck — CHANGES_REQUESTED** [review]
Review done with just a couple of comments.

I pushed the first handful of commits to decrease the PR size and potential future rebase work.

File: internal/daemon/logstashconfig/file.go [hunk 1]
```diff
@@ -51,29 +51,60 @@ func (f *File) parse() error {
...
+		case "add_field", "tags":
+			attrs = append(attrs, attr)
+		case "codec":
+			r.inputCodecs[id] = attr.String()
+		default:
+		}
+	}
+
 	// TODO: __lfv_input must reflect the actual input, that has been replaced, such that this input
 	// can be referenced in the test case configuration.
```
[the remark this answers is not in the document]
Comment (breml) [reply internal/daemon/logstashconfig/file.go:106]: You are right, this is implemented now.

File: internal/testcase/testcase.go [hunk 2]
```diff
@@ -32,6 +32,13 @@ type TestCaseSet struct {
 	// test case was read.
 	File string `json:"-" yaml:"-"`
 
+	// The unique ID of the input plugin in the tested configuration, where the
+	// test input is coming from. This is necessary, if a setup with multiple
+	// inputs is tested, which either have different codecs or are part of
+	// different pipelines.
+	// https://www.elastic.co/guide/en/logstash/7.10/plugins-inputs-file.html#plugins-inputs-file-id
+	InputPlugin string `json:"input_plugin" yaml:"input_plugin"`
```
[the remark this answers is not in the document]
Comment (breml) [reply internal/testcase/testcase.go:40]: Fixed, changed to snake case.

**Comment by breml** [comment]
@magnusbaeck I addressed your review feedback and updated the PR.

**Review by magnusbaeck — APPROVED** [review]

**Pull request merged**